[Yu Qian] iP

yuqiannemo opened the pull request:
### Nemo

> Just keep swimming -- Finding Nemo

Nemo helps you manage your tasks effortlessly, so you can focus on what truly matters. It’s:

- Text-based
- Simple to use
- ~~FAST~~ Blazingly FAST

Get started in just a few steps:

1. Download it [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/...).
2. Run it with a double-click.
3. Add your tasks.
4. Let Nemo handle the rest! 😉

And the best part? It’s **FREE**!

Features:

- [x] Task management
- [ ] Deadlines & scheduling (coming soon)
- [ ] Smart reminders (coming soon)

If you’re a Java programmer, you can also use Nemo to sharpen your skills. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
Download Nemo today and make productivity effortless! 😃

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yuqiannemo committed:
rename, greet and exit

yuqiannemo committed:
add echo

yuqiannemo committed:
add store and list feature

yuqiannemo committed:
add Task class

yuqiannemo committed:
add mark and unmark features

yuqiannemo committed:
add 3 types of tasks using inheritance

yuqiannemo committed:
add logic to support 3 tasks

yuqiannemo committed:
refine logic for handling tasks

yuqiannemo committed:
add A-TextUiTesting

yuqiannemo committed:
add NemoException

yuqiannemo committed:
use switch syntax for 3 tasks and handle exceptions

yuqiannemo committed:
handle exceptions for mark and unmark

yuqiannemo committed:
add delete feature

yuqiannemo committed:
merge delete feature with mark and unmark

yuqiannemo committed:
update tests

yuqiannemo committed:
use enums for command and task status

yuqiannemo committed:
update tests with delete use case

yuqiannemo committed:
Add tasks file

yuqiannemo committed:
Merge branch-Level-7 with task file implementation

yuqiannemo committed:
Change task time to LocalDate type

yuqiannemo committed:
Add logic for LocalDate

yuqiannemo committed:
Add nemo emoji

yuqiannemo committed:
bug fix: handle wrong input for event and deadline

yuqiannemo committed:
Ignore tasks.txt

yuqiannemo committed:
Merge branch-Level-8 with date implementation

yuqiannemo committed:
Add Ui class

yuqiannemo committed:
Add Storage class

yuqiannemo committed:
Add Parser class

yuqiannemo committed:
Add TaskList class

yuqiannemo committed:
Modify logic in the main code for OOP

yuqiannemo committed:
Move TaskStatus enums into Task class

yuqiannemo committed:
Add Command class and its subclasses

yuqiannemo committed:
Merge branch-MoreOOP for more OOP

yuqiannemo committed:
Divide classes into packages

yuqiannemo committed:
Merge branch-Packages with implementation of packages

yuqiannemo committed:
Add gradle support

yuqiannemo committed:
Fix typo in Ui.java

yuqiannemo committed:
Add JUnit Tests

yuqiannemo committed:
Correct JAR file name

yuqiannemo committed:
Add JavaDoc for Nemo.java

yuqiannemo committed:
Add JavaDoc for java files in nemo

yuqiannemo committed:
Add JavaDoc for command classes

yuqiannemo committed:
Add JavaDoc for task classes

yuqiannemo committed:
Reformat code

yuqiannemo committed:
Add find command

yuqiannemo committed:
JavaDoc implementation

yuqiannemo committed:
Obey coding standards

yuqiannemo committed:
Add Level-9 find command

yuqiannemo committed:
Add import

yuqiannemo committed:
Add checkStyle

yuqiannemo committed:
Add checkStyle

yuqiannemo committed:
Add echo GUI for Nemo

yuqiannemo committed:
Change Ui return type to String

yuqiannemo committed:
Modify logic to correctly respond

Taoseeker commented on src/main/java/nemo/Nemo.java:
```diff
@@ -0,0 +1,73 @@
+package nemo;
```
The coding standard LGTM

Taoseeker reviewed (commented):
Coding standard guidelines are closed followed.

yuqiannemo committed:
Add termination logic

yuqiannemo committed:
Add welcome message

yuqiannemo committed:
Replace Nemo image with a better one

yuqiannemo committed:
Add auto resize

yuqiannemo committed:
Add css

yuqiannemo committed:
Add logic for style-change based on commands

yuqiannemo committed:
Add background image

yuqiannemo committed:
Clean up code

yuqiannemo committed:
Change Dory image

yuqiannemo committed:
Modify java version manually

yuqiannemo committed:
Add GUI

yuqiannemo committed:
Add varargs in find command

yuqiannemo committed:
Add varargs

zechary28 commented on src/main/java/nemo/Storage.java:
```diff
@@ -0,0 +1,144 @@
+        } catch (IOException e) {
+            throw new NemoException("Error saving tasks: " + e.getMessage());
+        }
+    }
```
good job, very clean and readable

zechary28 commented on src/main/java/nemo/command/AddEventCommand.java:
```diff
@@ -0,0 +1,77 @@
+     * @throws NemoException If the input format is invalid or the dates are in the wrong format.
+     */
+    public AddEventCommand(String message) throws NemoException {
+        if (!message.contains("/from") || !message.contains("/to")) {
```
by De Morgan's theorem `message.contains("/from") && message.contains("/to")` i think would be more readable

zechary28 commented on src/main/java/nemo/command/AddEventCommand.java:
```diff
@@ -0,0 +1,77 @@
+        }
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws NemoException {
+        Event event = new Event(description, from, to);
+        tasks.add(event);
+        storage.save(tasks);
+        return ui.getTaskAddedMessage(event, tasks.size());
+    }
```
nice lgtm

zechary28 commented on src/main/java/nemo/Parser.java:
```diff
@@ -0,0 +1,55 @@
+        String[] messageArray = message.split(" ");
+        String commandStr = messageArray[0].toUpperCase();
+
+        switch (commandStr) {
+        case "LIST":
+            return new ListCommand();
+        case "BYE":
+            return new ExitCommand();
```
very neat and scalable, good job

zechary28 reviewed (commented):
Overall very well structured and neat, could not find any major issues :)))

yuqiannemo committed:
Add assertions

yuqiannemo committed:
Refactor code for Storage.java

Previous code is not of good readability.

Extract out several methods, including createTaskFile and loadTasks.

Doing so allows the load method to be more readable and also of single level of abstraction.

yuqiannemo committed:
Remove null string in StringBuilder initialization

yuqiannemo committed:
Refactor code for task classes

yuqiannemo committed:
Refactor code for command classes

Command classes output a lot of messages which are not organized into constants.

Extract these messages into constants, and some are shared among subclasses of Command.

By doing so, the readability of the code can be improved.

yuqiannemo committed:
Add assertion for TaskList.java

yuqiannemo committed:
Add Java Stream in Parser.java

yuqiannemo committed:
Merge pull request #1 from yuqiannemo/branch-A-Assertions

Add assertions

yuqiannemo committed:
Merge branch 'master' into branch-A-CodeQuality

yuqiannemo committed:
Merge branch 'master' into branch-A-Streams

yuqiannemo committed:
Merge pull request #2 from yuqiannemo/branch-A-CodeQuality

Improve the code quality

yuqiannemo committed:
Merge branch 'master' into branch-A-Streams

yuqiannemo committed:
Merge pull request #3 from yuqiannemo/branch-A-Streams

Add Java Stream

yuqiannemo committed:
Set up CI

yuqiannemo committed:
Set up CI

yuqiannemo committed:
Add new lines at the end

yuqiannemo committed:
Add reminders

yuqiannemo committed:
Make reminders non-repeat

yuqiannemo committed:
Stop scheduling service when stage is closed

yuqiannemo committed:
Add extension B-Reminders

yuqiannemo committed:
Fix reminder

yuqiannemo committed:
Rename boolean variable

yuqiannemo committed:
Add screenshot of the app

yuqiannemo committed:
Rewrite README.md

yuqiannemo committed:
Reformat code

yuqiannemo committed:
Remove unused comment
